### PR TITLE
add tasktemplate screens

### LINKED ISSUE #320: Task Template Screen
- [ ] TwoColumn layout based screen
- [ ] Personal Task Template Screen (navigable using `UserMenu`)
- [ ] Ward Task Template Screen (navigable over WardDetailsScreen)

Tasks - overview of Task Template in a wards

### PR DESCRIPTION
**Pull request by DasProffi**
closes #320

### CODE REVIEW

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: tasks/components/layout/TaskTemplateDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    switchToWardView: 'Stations Task Vorlagen',
+    taskTemplates: 'Task Vorlagen'
+  }
+}
+
+export type TaskTemplateDisplayProps = {
+  selectedID: string,
+  onSelectChange: (taskTemplate: TaskTemplateDTO | undefined) => void,
+  taskTemplates: TaskTemplateDTO[],
+  variant: 'wardTemplates' | 'personalTemplates'
```
[the remark this answers is not in the document]
Comment (DasProffi) [reply tasks/components/layout/TaskTemplateDisplay.tsx:38]: it is required for the button to switch the view, or what alternative do you suggest there

[the remark this answers is not in the document]
Comment (MaxSchaefer) [reply tasks/components/layout/TaskTemplateDisplay.tsx:38]: I slept on it, nothing came to mind :D

**Review by MaxSchaefer — DISMISSED** [review]

**Review by use-to — CHANGES_REQUESTED** [review]
image

Those breadcrumbs are wrong! For personal task templates we should do `Task Templates` or `Personal / Task Templates` or something like that.

Maybe switch to one-way navigation (personal task templates can't switch to ward task template directly) or pass the ward uuid as get url parameter.

**Review by use-to — CHANGES_REQUESTED** [review]
Task template notes are not updated on click.

**Comment by use-to** [comment]
Merging is caring 🍔

File: tasks/components/layout/TaskTemplateDisplay.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  switchToWardView: string,
+  taskTemplates: string
+}
+
+const defaultTaskTemplateDisplayTranslation = {
+  en: {
+    addNewTaskTemplate: 'Add new template',
+    switchToPersonal: 'Personal Task Templates',
+    switchToWardView: 'Ward Task Templates',
+    taskTemplates: 'Task Templates'
```
Comment (use-to) [inline tasks/components/layout/TaskTemplateDisplay.tsx:24]: We should reword this to: `Ward Task Templates`  or `Personal Task Templates`

**Review by use-to — CHANGES_REQUESTED** [review]

**Review by use-to — CHANGES_REQUESTED** [review]
Task Templates stays Task Templates in german for us. We decided that some days ago. ^^

File: tasks/components/layout/TaskTemplateDisplay.tsx (hunk 2 above)
Comment (DasProffi) [reply tasks/components/layout/TaskTemplateDisplay.tsx:24]: That's exactly what stands there, or did you mean the German translation too

Comment (use-to) [reply tasks/components/layout/TaskTemplateDisplay.tsx:24]: image

File: tasks/components/layout/TaskTemplateDisplay.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+}: PropsWithLanguage<TaskTemplateDisplayTranslation, TaskTemplateDisplayProps>) => {
+  const translation = useTranslation(language, defaultTaskTemplateDisplayTranslation)
+  const router = useRouter()
+  const { uuid } = router.query
+
+  return (
+    <div className={tw('py-4 px-6')}>
+      <div className={tw('flex flex-row items-center justify-between mb-4')}>
+        <Span type="subsectionTitle">
+          {variant === 'personalTemplates' ? translation.switchToPersonal : translation.switchToWardView}
```
Comment (use-to) [inline tasks/components/layout/TaskTemplateDisplay.tsx:57]: This is an anti-pattern. Rename the translation!

File: tasks/components/layout/TaskTemplateDisplay.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+            onClick={() => {
+              // TODO maybe replace with something else later
+              router.push(variant === 'personalTemplates' ? `/ward/${uuid}/templates` : '/templates').then()
+            }}
+            className={tw('flex flex-row gap-x-1 items-center')}
+          >
+            <LucideArrowLeftRight/>
+            {variant === 'personalTemplates' ? translation.switchToWardView : translation.switchToPersonal}
+          </Button>
+        </div>
```
Comment (use-to) [inline tasks/components/layout/TaskTemplateDisplay.tsx:70]: Display only on Ward is specified via get-params

**Review by use-to — CHANGES_REQUESTED** [review]

File: tasks/components/layout/TaskTemplateDisplay.tsx (hunk 3 above)
Comment (DasProffi) [reply tasks/components/layout/TaskTemplateDisplay.tsx:57]: Changing the name makes sense, because it's not accurate. What exactly is the antipattern here?

Comment (use-to) [reply tasks/components/layout/TaskTemplateDisplay.tsx:57]: Abusing translations is an antipattern.

Comment (DasProffi) [reply tasks/components/layout/TaskTemplateDisplay.tsx]: There are two separate translations; each should get its key (we had this conversation with @JannikWibker before)

So is the ternary operator the problem, or what exactly is it?

Comment (use-to) [reply tasks/components/layout/TaskTemplateDisplay.tsx]: `translation.switchToPersonal` is used as `translation.personalTitle` or something like that

Comment (DasProffi) [reply tasks/components/layout/TaskTemplateDisplay.tsx]: yes, so it's a problem of the translation key implying another use, so renaming it to `personalTaskTemplates` would solve this?

**Review by use-to — APPROVED** [review]

**Review by use-to — APPROVED** [review]

**Review by use-to — APPROVED** [review]
